### DISCUSSION TITLE
Templates and user editable views

### DISCUSSION THREAD

**Discussion by 7omas** [cpvalente/ontime#1121] [Q&A]
Some views were previously removed since they was no way to make it universal in a good way (PiP comes to mind but there are others..)

Created this issue so that it might be a discussion and reference point in implementing user views with templates. These templates would both allow specific views to be made but would also be a great starting point for anyone wanting to create their own and looked for examples of how that can be implemented.

Should this be a sub-directory that can be accessed from the Ontime url and is therefore a selectable directory on the system running Ontime. Or should this be local files that just reference the Ontime API? (Might then require manual editing of files to match the API IP address if the target usage in not on the same system)

**Comment by alex-Arc** [comment]
Hey @7omas 
There are is demo page at `http://<ontime-ip>:<ontime-port>/external/demo`
to show as a starting point how you can make your own
on windows it's located here: `AppData\Roaming\Ontime\external\demo`
anything you place in the `external` folder will be made available through the the ontime server
there is a bit more documentation here https://docs.getontime.no/features/custom-views/
That should be a good starting point

**Comment by cpvalente** [comment]
Thank you @7omas , as @alex-Arc mentioned, Ontime already provides a simple demo and infrastructure to distribute user code which I believe covers the case in your request.

The demo here is rather simple but it should be good enough to get you started. 

It is also worth noting that the views are made so that they can be easily consulted by someone who has minimal knowledge of react. 
We have traded convenience and verbosity for keeping the code simple and free of abstractions for the exact purpose of allowing users to understand what is happening

As far as having more demos here, this would have to be a user initiated effort.
If someone is making views I am happy to include them in Ontime or document their existence so that users can find it.
Otherwise, it is unlikely that we will be adding things here
